starknet_devnet/blueprints/postman.py

```diff
@@ -55,3 +56,30 @@ async def flush():
 
     result_dict = await state.starknet_wrapper.postman_flush()
     return jsonify(result_dict)
+
+
+@postman.route("/l1_to_l2", methods=["POST"])
+async def l1_to_l2():
+    """L1 to L2 message mock endpoint"""
+    request_json = request.json or {}
+    l1_contract_address = extract_hex_string(request_json, "l1_contract_address")
+    l2_contract_address = extract_hex_string(request_json, "l2_contract_address")
+    entry_point_selector = extract_hex_string(request_json, "entry_point_selector")
+    payload = extract_hex_string(request_json, "payload", to_int_array)
+    nonce = extract_hex_string(request_json, "nonce")
+
+    result = await state.starknet_wrapper.postman_l1_to_l2(
+        l1_contract_address=l1_contract_address,
+        l2_contract_address=l2_contract_address,
+        entry_point_selector=entry_point_selector,
+        payload=payload,
+        nonce=nonce,
+    )
+
+    try:
+        result_iterator = iter(result)
+        return jsonify({"execution_info_calldata": list(map(hex, result_iterator))})
+    except TypeError:
+        return jsonify(
+            {"execution_info_calldata": str(StarkErrorCode.INVALID_TRANSACTION)}
```

Comment:
I used calldata to be sure that invoking has actual data inside, but I can change it to txs

Reply:
So you want `mock_message_to_l2` to return `tx_handler.internal_tx`?

Reply:
ok so you want to return `tx_handler.internal_tx` instead of `tx_handler.execution_info.call_info.calldata` in `mock_message_to_l2()` is that correct?

Reply:
fixed in https://github.com/Shard-Labs/starknet-devnet/pull/365/commits/8f1e7bb41cb060144d981dd6b122638b828b103f